dashboard/app/models/user_level.rb

```diff
@@ -24,6 +26,10 @@ class UserLevel < ActiveRecord::Base
   belongs_to :level
   belongs_to :script
 
+  scope :attempted, -> { where.not(best_result: nil) }
+  scope :passed, -> { where('best_result >= ?', ActivityConstants::MINIMUM_PASS_RESULT) }
```

Comment:
are submitted, finished, and passing all also adjectives? in that case would 'perfect' work (and passing still be ok)? (I now see that perfect != best, which is interesting...)
Ok I see that it is already inconsistent, but I think that I would prefer that we stay at the same level of inconsistency instead of making it more inconsistent. Feel free to change the other names if you like your new ones better but that is probably more work than you want to do :)

This is the other existing place (where I saw perfect, and I see now also uses passed and not passing :p): https://github.com/code-dot-org/code-dot-org/blob/17d4cc9b6f56779760b587c897a5546f7e587330/dashboard/app/helpers/application_helper.rb#L51